connectors/sources/mysql.py

```diff
@@ -407,50 +396,46 @@ async def _connect(self, query, fetch_many=False, **query_kwargs):
         rows_fetched = 0
         cursor_position = 0
 
-        async with self._mysql_client.with_connection_pool() as connection_pool:
+        async with self._mysql_client() as client:
             while retry <= self.retry_count:
                 try:
-                    async with connection_pool.acquire() as connection:
-                        async with connection.cursor(
-                            aiomysql.cursors.SSCursor
-                        ) as cursor:
-                            await cursor.execute(formatted_query)
-
-                            if fetch_many:
-                                # sending back column names only once
-                                if yield_once:
-                                    yield [
-                                        f"{query_kwargs['table']}_{column[0]}"
-                                        for column in cursor.description
-                                    ]
-                                    yield_once = False
-
-                                # setting cursor position where it was failed
+                    async with client.connection.cursor(
```

Comment:
I kinda feel that you can encapsulate `connection.cursor` in the context manager.

Do you think you can also inline retry policy inside the client?

Reply:
Yes, that'll be inlined, when I split `_connect` up into separate methods 👍